osu.Game.Rulesets.Catch/Skinning/Legacy/LegacyHitExplosion.cs

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Sprites;
+using osu.Game.Rulesets.Catch.Objects;
+using osu.Game.Rulesets.Catch.UI;
+using osu.Game.Skinning;
+using osuTK;
+
+namespace osu.Game.Rulesets.Catch.Skinning.Legacy
+{
+    public class LegacyHitExplosion : CompositeDrawable, IHitExplosion
+    {
+        [Resolved]
+        private Catcher catcher { get; set; }
+
+        private const float catch_margin = (1 - Catcher.ALLOWED_CATCH_RANGE) / 2;
+
+        private readonly Sprite explosion1;
+        private readonly Sprite explosion2;
+
+        public LegacyHitExplosion()
+        {
+            Anchor = Anchor.BottomCentre;
+            Origin = Anchor.BottomCentre;
+            RelativeSizeAxes = Axes.Both;
+            Scale = new Vector2(0.4f);
+
+            InternalChildren = new[]
+            {
+                explosion1 = new Sprite
+                {
+                    Anchor = Anchor.BottomCentre,
+                    Origin = Anchor.CentreLeft,
+                    Alpha = 0,
+                    Blending = BlendingParameters.Additive,
+                    Rotation = -90
+                },
+                explosion2 = new Sprite
+                {
+                    Anchor = Anchor.BottomCentre,
+                    Origin = Anchor.CentreLeft,
+                    Alpha = 0,
+                    Blending = BlendingParameters.Additive,
+                    Rotation = -90
+                }
+            };
+        }
+
+        [BackgroundDependencyLoader]
+        private void load(SkinManager skins)
+        {
+            var defaultLegacySkin = skins.DefaultLegacySkin;
+
+            explosion1.Texture = defaultLegacySkin.GetTexture("scoreboard-explosion-2");
+            explosion2.Texture = defaultLegacySkin.GetTexture("scoreboard-explosion-1");
+        }
+
+        public void Animate(HitExplosionEntry entry)
+        {
+            if (entry == null)
+                return;
+
+            Colour = entry.ObjectColour;
+
+            using (BeginAbsoluteSequence(entry.LifetimeStart))
+            {
+                float halfCatchWidth = catcher.CatchWidth / 2;
+                float explosionOffset = Math.Clamp(entry.Position, -halfCatchWidth + catch_margin * 3, halfCatchWidth - catch_margin * 3);
+
+                if (!(entry.HitObject is Droplet))
+                {
+                    float scale = Math.Clamp(entry.JudgementResult.ComboAtJudgement / 200f, 0.35f, 1.125f);
+
+                    explosion1.Scale = new Vector2(1, 0.9f);
+                    explosion1.Position = new Vector2(explosionOffset, 0);
+
+                    explosion1.FadeOutFromOne(300);
+                    explosion1.ScaleTo(new Vector2(20 * scale, 1.1f), 160, Easing.Out);
```

Comment:
Is there a reason you've used 20 here? Stable seems to use 16 instead going by the code alone.

Reply:
The reason is mostly that the original PR used 20. But I'll double check whether that was correct for sure.

Reply:
I've stared at the sizings for a bit and this is now 16 again with b84f238 - but I *also* nudged the cryptic "scale" from `0.4f` to `0.5f`...  Here's a freeze-frame image comparison to explain why (recommend opening in tabs and tabbing back and forth):

| stable | this PR, initially | this PR, now |
| :-: | :-: | :-: |
| ![screenshot025](https://user-images.githubusercontent.com/20418176/129266790-bfaa87ae-34f0-430f-893f-d7bdc2c4279c.jpg) | ![osu_2021-08-12_21-51-28](https://user-images.githubusercontent.com/20418176/129266704-b456b6e8-a990-449d-87cc-789903479d24.jpg) | ![osu_2021-08-12_22-31-18](https://user-images.githubusercontent.com/20418176/129266717-5e89e224-0635-4ded-a8c2-681d79a0dae7.jpg) |

Note the edge blur in particular - the `0.4f` was squishing things too much it looks.

I'm... not actually sure where the `0.5f` is coming from either. I'll look into it on request, but I'm sort of hoping that this can be accepted as "close enough" visually.